packages/edit-post/src/components/header/style.scss

```diff
@@ -33,7 +34,14 @@
 
 .edit-post-header__toolbar {
 	display: flex;
-	flex-grow: 1;
+	// Allow this area to shrink to fit the toolbar buttons.
+	flex-shrink: 8;
+	// Take up the space of the toolbar so it can be justified to the left side of the toolbar.
+	flex-grow: 3;
+	// Hide the overflow so flex will limit its width. Block toolbar will allow scrolling on fixed toolbar.
```

Comment:
We talked about adding a test that the toolbar is scrollable...

Reply:
Yup! I'm still planning to add that.

Reply:
Tricky one to figure out how to do it in a sensible and reliable way. https://github.com/WordPress/gutenberg/pull/55787/commits/62a4591158d0cbb306d3c6b3c5b7c87ecc48e06f